queue.go

```diff
@@ -459,6 +460,26 @@ func (queue *redisQueue) move(from, to string, max int64) (n int64, error error)
 	return n, nil
 }
 
+// Drain removes and returns 'count' elements from the queue. In case of an error,
+// Drain return all elements removed until the error occurred and the error itself.
+func (queue *redisQueue) Drain(count int64) ([]string, error) {
+	var (
+		n   int64
+		err error
+	)
+	out := make([]string, 0, count)
+
+	for n = 0; n < count; n++ {
+		val, err := queue.redisClient.RPop(queue.readyKey)
```

Comment:
Have you considered popping multiple elements at once. That should be much more efficient. Maybe not all `count` at once, but batches of maybe 100 per RPop?
https://redis.io/commands/rpop/